components/alert/alert-toast.js

```diff
@@ -0,0 +1,183 @@
+import '../button/button-icon.js';
+import '../button/button-subtle.js';
+import '../colors/colors.js';
+import './alert.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+const states = {
+	CLOSED: 'closed', // the toast is closed
+	CLOSING: 'closing', // the close animation is running
+	PREOPENING: 'preopening', // a pause before running the opening animation because transitions won't run when changing from 'diplay: none' to 'display: block'
+	OPENING: 'opening', // the opening animation is running
+	OPEN: 'open' // the toast is open
+};
+
+class AlertToast extends LitElement {
+
+	static get properties() {
+		return {
+			buttonText: { type: String, attribute: 'button-text' },
+			hideCloseButton: { type: Boolean, attribute: 'hide-close-button' },
+			noAutoClose: { type: Boolean, attribute: 'no-auto-close' },
+			open: { type: Boolean, reflect: true },
+			subtext: { type: String },
+			type: { type: String, reflect: true },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+
+			.d2l-alert-toast-container {
+				border-radius: 0.3rem;
+				bottom: 1.5rem;
+				box-shadow: 0 0.1rem 0.6rem 0 rgba(0,0,0,0.10);
+				display: none;
+				left: 0;
+				margin: 0 auto;
+				max-width: 600px;
+				position: fixed;
+				right: 0;
+				width: 100%;
+				z-index: 10000;
+			}
+
+			:host(:not([_state="closed"])) .d2l-alert-toast-container {
+				display: block;
+			}
+
+			:host([_state="opening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				transition-duration: 250ms;
+				transition-property: transform, opacity, visibility;
+				transition-timing-function: ease-in;
+			}
+
+			:host([_state="preopening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				opacity: 0;
+				transform: translateY(0.5rem);
+				visibility: hidden;
```

Comment:
Are you sure you need `visibility`?  I _think_ `opacity` should be enough.  We do this sort of thing elsewhere, ex.  [visible-on-ancestor](https://github.com/BrightspaceUI/core/blob/master/mixins/visible-on-ancestor-mixin.js#L8), and there might be one or two other cases.  If looking at VoA, ignore the `!important`... it is being used to override the host in event it is setting those properties with high specificity.

Also, seems like there might be more states than what you really need, but whatever is most understandable.  :)

Reply:
`opacity` is likely enough, `visibility` was carried through from the Polymer version. I'll give it a go.

Regarding redundant states, I could combine `opening` and `open`.
I experimented with it a bit and found that:
1. Made the code slightly more difficult to understand because `closing` had no corresponding `opening` state
1. There was no easy way to detect when the alert finished opening to do things like change focus, announce text, or start the auto-close timer.
1. I had to replace it with the `reduce-motion` media query anyway.